Producer-Consumer problem/src/org/webonise/producerconsumerproblem/Consumer.java

```diff
@@ -0,0 +1,55 @@
+package org.webonise.producerconsumerproblem;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+
+public class Consumer implements Runnable{
+    private final ArrayList<Integer> sharedItem;
+    private int size;
+    public Consumer(ArrayList<Integer>sharedItem,int size){
+        // sets values of fields.
+        this.sharedItem=sharedItem;
+        this.size=size;
+    }
+    @Override
+    public void run() {
+        while(true){
+            try {
+                // Consumer consumes Product.
+                System.out.println("Consumed:"+consume());
+                display();
+                Thread.sleep(50);
+            }catch(Exception e){
+                e.printStackTrace();
+            }
+        }
+    }
+    private int consume()throws Exception{
+            // Wait if size of sharedItem is empty.
+            while(sharedItem.isEmpty()){
+                // Used Synchronized block to achieve mutual exclusion.
+                synchronized (sharedItem){
```

Comment:
Why do you have `synchronized` code block here? I do not see any read or write operation on `sharedItem` in this block.